### PR TITLE
[SPARK-19130][SPARKR] Support setting literal value as column implicitly

### PR DESCRIPTION
**Pull request by felixcheung**
## What changes were proposed in this pull request?

```
df$foo <- 1
```

instead of
```
df$foo <- lit(1)
```

## How was this patch tested?

unit tests

### CODE REVIEW

**Comment by rxin** [comment]
This is pretty useful!

File: R/pkg/R/DataFrame.R [hunk 1]
```diff
@@ -1727,14 +1727,21 @@ setMethod("$", signature(x = "SparkDataFrame"),
...
 #' @aliases $<-,SparkDataFrame-method
 #' @note $<- since 1.4.0
 setMethod("$<-", signature(x = "SparkDataFrame"),
           function(x, name, value) {
-            stopifnot(class(value) == "Column" || is.null(value))
+            if (class(value) != "Column" && !is.null(value)) {
+              if (isAtomicLengthOne(value)) {
+                value <- lit(value)
+              } else {
+                stop("value must be a Column, Literal value as atomic in length of 1, or NULL")
```
Comment (shivaram) [inline R/pkg/R/DataFrame.R]: nit - it must be `literal` (no caps ?)

**Review by shivaram — COMMENTED** [review]
Thanks @felixcheung - LGTM.

File: R/pkg/R/DataFrame.R (hunk 1 above)
Comment (felixcheung) [inline R/pkg/R/DataFrame.R]: I was thinking org.apache.spark.sql.catalyst.expressions.Literal as the type, but I guess i can use the lower case term too

**Comment by shivaram** [comment]
Merging this to master, branch-2.1

**Comment by shivaram** [comment]
I can't update JIRA as it seems to be down right now ? (Getting a 503) - Anyways I'll check again in a few hours

**Pull request closed without merging**